lib/resty/healthcheck.lua

```diff
@@ -899,17 +899,16 @@ function checker:run_single_check(ip, port, hostname, hostheader)
   if self.checks.active.type == "https" then
     local https_sni, session, err
     https_sni = self.checks.active.https_sni or hostheader or hostname
+
     if self.ssl_cert and self.ssl_key then
-      session, err = sock:tlshandshake({
-        verify = self.checks.active.https_verify_certificate,
-        client_cert = self.ssl_cert,
-        client_priv_key = self.ssl_key,
-        server_name = https_sni
-      })
-    else
-      session, err = sock:sslhandshake(nil, https_sni,
-                                     self.checks.active.https_verify_certificate)
+      ok, err = sock:setclientcert(self.ssl_cert, self.ssl_key)
```

Comment:
Idea: what if we used feature detection (some combo of `if type(sock.setclientcert) == "function"` and `if type(sock.tlshandshake) == "function"`) instead of a hard-coded change? In my mind then we would have the flexibility to merge+release this now rather than having one more transitive dependency to carefully update in lockstep with Kong. Replacing the feature detection with the hard-coded behavior could be left as a TODO once OpenResty+Kong are all on the same page, socket feature-wise.